### DISCUSSION TITLE
SignalK TypeError: Cannot read properties of null (reading 'matches')

### DISCUSSION THREAD

**Discussion by toastygm** [bareboat-necessities/lysmarine_gen#349] [General]
Every time I try to install anything in SignalK, I get the following error:

TypeError: Cannot read properties of null (reading 'matches')

I saw that this problem was supposed to be fixed back in 2022, but the problem is now happening to me.  Any ideas on what could be the problem.

This is a new installation of BBN from August 2023.

**Comment by mgrouch** [comment]
What is output of

cat  /etc/bbn-build

?


Did you perform any upgrade of SignalK after you loaded BBN image?

Can you give an exact example of what you are trying to install so I could reproduce the issue?

Thanks

**Comment by toastygm** [comment]
```
$ cat /etc/bbn-build
2023-07-23 18:29:49+00:00
```

I am attempting to install `signalk-postgsail`, and I get "failed" every time.

Same problem no matter what I try to install, upgrade, or remove.  Even upgrade of signalk itself fails.

**Comment by mgrouch** [comment]
Upgrade of signalK  itself is not supported. There is a way to do it on command line but it's not really the procedure for average user.

I've tried updating on 2023-08-17 and it worked.
![Screenshot_2023-10-10_15-22-31](https://github.com/bareboat-necessities/lysmarine_gen/assets/8263013/3d553c33-b071-4707-b07b-f99b90267933)

**Comment by mgrouch** [comment]
I've tried installing signalk-postgsail on 2023-08-17 and it worked for me.

**Comment by mgrouch** [comment]
![Screenshot_2023-10-10_15-33-58](https://github.com/bareboat-necessities/lysmarine_gen/assets/8263013/0ced36aa-17e1-4bd6-a6fa-1f9155e5e6bc)

**Comment by toastygm** [comment]
Running the following commands ended up fixing it for me:

```
# use pnpm instead of npm
sed -i 's#('"'npm',#\('pnpm'"',#' /usr/lib/node_modules/signalk-server/lib/modules.js

# SignalK fix for pnpm
sed -i -e s/--save"'",/"--save-prod'",/g /usr/lib/node_modules/signalk-server/lib/modules.js
```
These are from the latest version of https://github.com/bareboat-necessities/lysmarine_gen/blob/bullseye/install-scripts/4-server/04-signalk.sh
